x-pack/plugins/security_solution/server/endpoint/services/actions/action_files.ts

```diff
@@ -133,3 +135,84 @@ const doesFileHaveChunks = async (
 
   return Boolean((chunks.hits?.total as SearchTotalHits)?.value);
 };
+
+/**
+ * Returns subset of fileIds that don't have any file chunks
+ *
+ * @param esClient
+ * @param fileIds
+ * @returns fileIds
+ */
+export async function fileIdsWithoutChunks(
+  esClient: ElasticsearchClient,
+  fileIds: string[]
+): Promise<string[]> {
+  const noChunkFileIds = new Set(fileIds);
+
+  const chunks = await esClient.search<{ bid: string }>({
+    index: FILE_STORAGE_DATA_INDEX,
+    body: {
+      query: {
+        terms: {
+          'bid.keyword': fileIds,
+        },
+      },
+      _source: ['bid'],
+    },
+  });
```

Comment:
I don't think this will do because each file can have 1 or more chunks and since you are not using page size, I think the default is 10 per page. You can probably just pull out 10k rows (use the `ACTIONS_SEARCH_PAGE_SIZE` const maybe? ([here](https://github.com/elastic/kibana/blob/6a6b1eeafab77e61a926e19294d71669571b2dde/x-pack/plugins/security_solution/server/endpoint/services/actions/constants.ts#L16))

That being said - we don't actually need to get out all of the chunks for all of the file IDs. We only need to know that each file has at least one chunk. The good thing is that the last chunk for each file has the property `last: true`, so you should use that in the query here (you still need to paginate the data or pull in 10k)